sdk/src/types/block/semantic/mod.rs

```diff
@@ -240,19 +272,63 @@ impl<'a> SemanticValidationContext<'a> {
                         Some(output.features()),
                     )
                 }
-                Output::Account(output) => (output.amount(), output.mana(), None, Some(output.features())),
+                Output::Account(output) => {
+                    if output.features().block_issuer().is_some() {
+                        let account_id = output.account_id_non_null(&OutputId::new(self.transaction_id, index as u16));
+
+                        if self.commitment_context_input.is_none() {
+                            return Ok(Some(TransactionFailureReason::BlockIssuerCommitmentInputMissing));
+                        }
+                        if !self.bic_context_inputs.contains(&account_id) {
+                            return Ok(Some(TransactionFailureReason::BlockIssuanceCreditInputMissing));
+                        }
+                        let entry = self.block_issuer_mana.entry(account_id).or_default();
+
+                        entry.1 = entry.1.checked_add(output.mana()).ok_or(Error::CreatedManaOverflow)?;
+
+                        if let Some(allotment) = self.transaction.allotments().get(&account_id) {
+                            entry.1 = entry
+                                .1
+                                .checked_add(allotment.mana())
+                                .ok_or(Error::CreatedManaOverflow)?;
+                        }
+                    }
+
+                    (output.amount(), output.mana(), None, Some(output.features()))
+                }
                 Output::Anchor(_) => return Err(Error::UnsupportedOutputKind(AnchorOutput::KIND)),
                 Output::Foundry(output) => (output.amount(), 0, output.native_token(), Some(output.features())),
                 Output::Nft(output) => (output.amount(), output.mana(), None, Some(output.features())),
                 Output::Delegation(output) => (output.amount(), 0, None, None),
             };
 
-            if let Some(sender) = features.and_then(|f| f.sender()) {
+            if let Some(sender) = features.and_then(Features::sender) {
                 if !self.unlocked_addresses.contains(sender.address()) {
                     return Ok(Some(TransactionFailureReason::SenderFeatureNotUnlocked));
                 }
             }
 
+            // TODO: should be done only towards block issuers
```

Comment:
Will you do it in this PR? Having a comment here which mentions that this is adding the `Account Out_locked`/Mana locked for an account would also be good

Reply:
Will probably do it in this PR yeah, still thinking what the best way is

Reply:
Done